lib/protobuf/nats/client.rb

```diff
@@ -13,8 +14,85 @@ def close_connection
         # no-op (I think for now), the connection to server is persistent
       end
 
+      def self.subscription_key_cache
+        @subscription_key_cache ||= {}
+      end
+
       def send_request
-        # after method execution the "response" should be in @response_data and should be a string of bytes
+        retries ||= 3
+
+        setup_connection
+        request_options = {:timeout => 60, :ack_timeout => 5}
+        @response_data = nats_request_with_two_responses(cached_subscription_key, @request_data, request_options)
+        parse_response
+      rescue ::NATS::IO::Timeout
+        # Nats response timeout.
+        retry if (retries -= 1) > 0
```

Comment:
If we send a message and servers are offline or full (drops a message), then we will always wait until the ACK timeout. Unlike ZMQ, we don't know if a server is listening to us or not, so we'll at least have the ACK timeout to prevent this firing without sleeps. This why we have an ACK timeout btw.. because if not, a server failure could leave an RPC client waiting for 60 seconds which is horrifically slow.